src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs

```diff
@@ -98,9 +98,10 @@ public override int GetHashCode()
                 return this.prefix.GetHashCode() ^ this.localname.GetHashCode();
             }
 
-            public int GetNSHashCode(SecureStringHasher hasher)
+            public int GetNSHashCode()
             {
-                return hasher.GetHashCode(this.namespaceUri) ^ hasher.GetHashCode(this.localname);
+                // ValueTuple uses HashCode.Combine with the randomized string.GetHashCode as inputs
```

Comment:
Would it be better to do `HashCode.Combine(namespaceUri, localname)` directly here? Avoids layers of indirection.

Reply:
#31784